File: plugins/shelly/src/main/java/eu/geekhome/shellyplugin/ShellyAdapterFactory.java
package eu.geekhome.shellyplugin;

import eu.geekhome.services.hardware.HardwareAdapter;
import eu.geekhome.services.hardware.HardwareAdapterFactory;
import eu.geekhome.services.mqtt.MqttBrokerService;

import java.util.ArrayList;
import java.util.List;

class ShellyAdapterFactory implements HardwareAdapterFactory {

    public static String ID = "SHELLY";

    private final MqttBrokerService _mqttBroker;

    ShellyAdapterFactory(MqttBrokerService mqttBroker) {
        _mqttBroker = mqttBroker;
    }

    @Override
    public List<HardwareAdapter> createAdapters() {
        ArrayList<HardwareAdapter> result = new ArrayList<>();
        ShellyAdapter adapter = new ShellyAdapter(_mqttBroker);
        result.add(adapter);

        return result;
    }

    @Override
    public String getId() {
        return ID;
    }
}

File: plugins/shelly/src/main/java/eu/geekhome/shellyplugin/ShellyAdapterFactory.kt
package eu.geekhome.shellyplugin

import eu.geekhome.domain.mqtt.MqttBrokerService
import eu.geekhome.domain.hardware.HardwareAdapterFactory
import eu.geekhome.domain.hardware.HardwareAdapter
import eu.geekhome.domain.langateway.LanGatewayResolver
import java.util.ArrayList

internal class ShellyAdapterFactory(
    override val owningPluginId: String,
    private val mqttBroker: MqttBrokerService,
    private val lanGatewayResolver: LanGatewayResolver) :
    HardwareAdapterFactory {

    override fun createAdapters(): List<HardwareAdapter<*>> {
        val result = ArrayList<HardwareAdapter<*>>()
        val adapter = ShellyAdapter(owningPluginId, mqttBroker, lanGatewayResolver)
        result.add(adapter)
        return result
    }
}